go-apps/meep-loc-serv/server/loc-serv.go

```diff
@@ -1112,30 +1127,52 @@ func zonalTrafficSubPost(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintf(w, string(jsonResponse))
 }
 
-func zonalTrafficSubPutById(w http.ResponseWriter, r *http.Request) {
+func zonalTrafficSubPut(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
 	vars := mux.Vars(r)
 
-	var response ResponseZonalTrafficSubscription
-	zonalTrafficSub := new(ZonalTrafficSubscription)
-	response.ZonalTrafficSubscription = zonalTrafficSub
+	var response InlineResponse20017
 
+	var body Body8
 	decoder := json.NewDecoder(r.Body)
-	err := decoder.Decode(&zonalTrafficSub)
+	err := decoder.Decode(&body)
 	if err != nil {
 		log.Error(err.Error())
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	zonalTrafficSub := body.ZonalTrafficSubscription
+
+	if zonalTrafficSub == nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	subsIdParamStr := vars["subscriptionId"]
+
+	selfUrl := strings.Split(zonalTrafficSub.ResourceURL, "/")
+	subsIdStr := selfUrl[len(selfUrl)-1]
```

Comment:
resource url is a mandatory string parameter... so it should be present... and on top of that will be allocated since it is a string, not a pointer.. so nothing to change here